### PR TITLE
add aarch64 on Cmake architecture

### PR DESCRIPTION
**Pull request by scorpionzezz**
based on #726  :+1:

### CODE REVIEW

File: cmake/modules/DetectArchitecture.cmake [hunk 1]
```diff
@@ -11,6 +11,8 @@ elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "arm32")
     set(ARM32 ON)
 elseif (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "arm64")
```
Comment (Wenzel) [inline cmake/modules/DetectArchitecture.cmake]: I think you can replace the `arm3264` strings by `aarch32/64`.

Comment (scorpionzezz) [reply cmake/modules/DetectArchitecture.cmake]: hmmm true, looks like it refers to the same thing (based on: https://unix.stackexchange.com/questions/461179/what-is-the-difference-between-different-implemetation-of-arm64-aarch64-for-linu)

Comment (scorpionzezz) [reply cmake/modules/DetectArchitecture.cmake]: done :+1:

Comment (Wenzel) [reply cmake/modules/DetectArchitecture.cmake]: maybe @tklengyel knows more about ARM ?

Comment (scorpionzezz) [reply cmake/modules/DetectArchitecture.cmake]: true, I tried with my Hikey960 board and it works, maybe @tklengyel  could try with the other boards.

**Comment by tklengyel** [comment]
I got the following error on my cubietruck:

```
CMake Error at cmake/modules/DetectArchitecture.cmake:15 (message):
  Undetect architecture armv7l to be configured in config.h.in
Call Stack (most recent call first):
  CMakeLists.txt:26 (include)


-- Configuring incomplete, errors occurred!

**Comment by scorpionzezz** [comment]
Then I guess we need to add arm7l as ARM32 ?

**Comment by tklengyel** [comment]
Looks like it

**Comment by scorpionzezz** [comment]
updated

**Comment by tklengyel** [comment]
I'm still getting the same error. You seem to mistyped `armv7l` missing the v from it.

**Comment by scorpionzezz** [comment]
ups, sorry, my bad, fixed :)

**Comment by tklengyel** [comment]
OK, it gets passed the cmake phase. Unfortunately either though it detects now that it's building on/for ARM, it still tries to build the Bareflank driver which of course fails. It would be nice of the cmake configuration automatically disabled that (as the configure script did).

**Pull request merged**